### PR TITLE
Send groupbuddy events

### PR DESCRIPTION
**Pull request by galfert**
Groupbuddy events are presence updates for people in a chat room.

I tried adding a test for it by defining an input stanza and the expected output, but the test fails because of a timeout.

@silverbucket: any idea what I'm doing wrong there?

### CODE REVIEW

File: test/incoming-data.js [hunk 1]
```diff
@@ -40,5 +40,26 @@ module.exports = [
         '@type': 'person'
       }
     }
+  },
+  {
+    name: 'presence-4',
```
Comment (skddc) [inline test/incoming-data.js]: That's not very descriptive?

**Comment by galfert** [comment]
> I tried adding a test for it by defining an input stanza and the expected output, but the test fails because of a timeout.

@silverbucket: any idea what I'm doing wrong there?

**Comment by silverbucket** [comment]
@galfert the tests here aren't passing. looks like the one you added is timing out?

**Comment by skddc** [comment]
@silverbucket Maybe read @galfert's comments in here?

**Comment by silverbucket** [comment]
Oh weird, totally missed that comment. Checking now.

**Comment by silverbucket** [comment]
@galfert This is because the incoming stanza tests just call the generic global stanza handler, which is what simple-xmpp calls for most things (there are only a few named listeners). Since the mock doesn't run the data through simple-xmpp (might be too complicated to add that into the route) perhaps it would be better to make a collection of named functions which are called from each of the stanza detection if/else's and, in addition, from an appropriate named listener handler. 

This way we can ensure the same code would be used in either case. What do you think?

see: https://github.com/sockethub/sockethub-platform-xmpp/blob/b957d4155e6cf5d8c69292c5654e1f8e9c31cea7/test/xmpp-incoming-suite.js#L65

**Comment by silverbucket** [comment]
Or we can specify another property on the incoming-data.js file, which indicates the listener handler to call if not `stanza`. 

```
{
    name: 'presence-4',
    handler: 'groupbuddy',
    input: '<presence to="hermes@5apps.com/hyperchannel" from="test@muc.5apps.com/greg" xmlns:stream="http://etherx.jabber.org/streams"><c ver="d2rgtMP0QRwWPU4dGU5DEFz5ZmM=" hash="sha-1" node="http://conversations.im" xmlns="http://jabber.org/protocol/caps"/><x xmlns="http://jabber.org/protocol/muc#user"><item role="moderator" affiliation="owner"/></x></presence>',
    output: {
      '@type': 'update',
      actor: {
        '@id': 'test@muc.5apps.com/greg',
        '@type': 'person',
        displayName: 'greg'
      },
      target: {
        '@id': 'test@muc.5apps.com',
        '@type': 'room'
      },
      object: {
        '@type': 'presence',
        status: null,
        presence: 'online'
      }
    }
}
```

The in the `xmpp-incoming-suite.js' (https://github.com/sockethub/sockethub-platform-xmpp/blob/b957d4155e6cf5d8c69292c5654e1f8e9c31cea7/test/xmpp-incoming-suite.js#L65) something like:
```javascript
env.platform.__listeners[entry.function || "stanza"].apply({ scope: env.session }, [ stanza ]);
```

**Comment by galfert** [comment]
Ok, I changed the test suite so that the incoming-data.js can define a `handler` and the `input` can either be a stanza string or an array of parameters for the handler.

**Comment by silverbucket** [comment]
Awesome, thanks for the updated @galfert, merging

**Pull request merged**